types.go

```diff
@@ -71,12 +71,33 @@ type Table[Obj any] interface {
 	// Prefix searches the table by key prefix.
 	Prefix(ReadTxn, Query[Obj]) (iter Iterator[Obj], watch <-chan struct{})
 
-	// DeleteTracker creates a new delete tracker for the table.
-	//
-	// It starts tracking deletions performed against the table from the
-	// current revision. A WriteTxn against the target table is required to
-	// add the tracker to the table.
-	DeleteTracker(txn WriteTxn, trackerName string) (*DeleteTracker[Obj], error)
+	// EventIterator returns an iterator for events happening on the table.
+	// This uses the revision index to iterate over the objects in the order
+	// they have changed. If 'includeDeleted' is true, then this registers a tracker
+	// to hold onto deleted objects until they are observed.
```

Comment:
If you want to go ahead with the `ReadTxn` then this should at least mention that if `includeDeleted` then you need to pass it a `WriteTxn` instead.

Reply:
You don't need to pass it a WriteTxn.